### ISSUE TITLE
Missing data sources

### ISSUE THREAD

**Issue by aistisdev** [al3xsh/rolling-element-bearing-fault-detection#4]
Hello.

I have read the paper and now am trying to get a better understanding (I am not an expert in this field) of how the problem is being solved, by reproducing the results in the paper. 
So far I solved the dependency issues where the imports work, but now am stuck at figuring out the data being used for the experiments.

**Description** 
Running [train_rnn_wdcnn_loads.py](https://github.com/al3xsh/rolling-element-bearing-fault-detection/blob/main/keras/rnn-wdcnn/train_rnn_wdcnn_loads.py) file I am not able to continue at [args = parser.parse_args()](https://github.com/al3xsh/rolling-element-bearing-fault-detection/blob/main/keras/rnn-wdcnn/train_rnn_wdcnn_loads.py#LC79), since it is not clear what kind of data has to be in the [../../data](https://github.com/al3xsh/rolling-element-bearing-fault-detection/blob/main/keras/rnn-wdcnn/experiments/experiment_1_example.txt) path. 

**Setup** 
I am using these dependencies:
- sklearn
- pandas
- keras==2.3.1
- tensorflow==2.2.0
- matplotlib

In the paper it is stated that the data from [cwru](https://csegroups.case.edu/bearingdatacenter/pages/download-data-file) was augmented, so it's currently not clear to me what files are needed for each model to run and how they were normalized and augmented. Would it be possible to provide this information, so i could reproduce the needed data files?

**Comment by al3xsh** [comment]
Hi,

In section "4.1. Data Description" we describe the use of sliding window data augmentation. This is actually taken care of by using the cwru_data_loader.py file (in keras/utils).

The data directory contains:

'48k_drive_end_fault' 1hp, 2hp, 3hp (for all listed fault conditions)
'48k_normal_baseline' 1hp, 2hp, 3hp

I'll commit my data_downloader script later today which will fetch and format all the CWRU data files. My notes say that I made some changes to:

48k_normal_baseline/2hp/normal_2.mat
48k_drive_end_fault/1hp/ir014_1.mat
48k_drive_end_fault/3hp/ir021_3.mat

as they contained errors - but unhelpfully I haven't recorded what those errors were! I seem to remember these were to do with the various variable naming within the CWRU files - I'll have to check when I get some time (I'll leave this issue open until then).

Regards,

Alex

**Comment by aistisdev** [comment]
I have made myself a little script to pull and rename the needed files, and now have this type of hierarchy:

![ss](https://user-images.githubusercontent.com/62371501/121688424-1cc8e880-cacc-11eb-82e7-029e4401018c.png)

However i have noticed that outer race fault files are named as "or007c" (only ending with c) in this [dictionary](https://github.com/al3xsh/rolling-element-bearing-fault-detection/blob/main/keras/utils/cwru_data_loader.py#LC65), from which i assume that only outer race centered files are used, but in the answer above you wrote - "'48k_drive_end_fault' 1hp, 2hp, 3hp (for all listed fault conditions)". Am i misunderstanding what "all listed fault conditions" means here?

**Comment by al3xsh** [comment]
Hi @aistisdev, you're right - it's just the orxxx@6 data that is used (the other positions weren't available for 0.014" fault conditions).

**Comment by al3xsh** [comment]
@aistisdev, also I've just had a look at the original data to see what I meant by corrections. The corrections I made were:

normal_2.mat (removed the variables in bold)
**X098_DE_time, X098_FE_time**, X099_DE_time, X099_FE_time

1hp/ir_014.mat (removed variables in bold)
**X217_DE_time**, X175_DE_time, X175_FE_time, X175RPM, **X217**

2hp/ir021.mat (removed variables in bold)
X217RPM, **X215_DE_time, X215_FE_time, X215RPM**, X217_DE_time, X217_FE_time

The original matlab files seem to have had some other variables snuck in to them and these confused the cwru_data_loader.py functionality :).

Also, I've now uploaded my data downloading scripts for reference :) see 01a6b302bb3c6ab036a5556bccfd36808427b463

Hope this helps,

Regards,

Alex

**Comment by al3xsh** [comment]
Closing now - hopefully solved :)
